### PR TITLE
Adds textured triangle support to the debug renderer

### PR DESCRIPTION
**Pull request by jankrassnigg**
Also fixes a memory leak and adds sample code to the sample game plugin.

### CODE REVIEW

File: Code/Engine/RendererCore/Debug/Implementation/DebugRenderer.cpp [hunk 1]
```diff
@@ -102,7 +103,7 @@ namespace
     }
 
     ezUInt64 m_uiLastRenderedFrame;
-    PerContextData* m_pData[2];
+    ezUniquePtr<PerContextData> m_pData[2];
```
Comment (jankrassnigg) [inline Code/Engine/RendererCore/Debug/Implementation/DebugRenderer.cpp:106]: This was a memory leak, and when you used one of the 'textured' calls (e.g. the textured Draw2DRectangle variant), D3D would complain about live GPU objects at shutdown.

File: Code/Engine/RendererCore/Debug/Implementation/DebugRenderer.cpp [hunk 2]
```diff
@@ -386,7 +388,7 @@ void ezDebugRenderer::Draw2DLines(const ezDebugRendererContext& context, ezArray
 }
 
 //static
-void ezDebugRenderer::DrawCross(const ezDebugRendererContext& context, const ezVec3& globalPosition, float fLineLength, const ezColor& color)
+void ezDebugRenderer::DrawCross(const ezDebugRendererContext& context, const ezVec3& globalPosition, float fLineLength, const ezColor& color, const ezTransform& transform /*= ezTransform::IdentityTransform()*/)
```
Comment (jankrassnigg) [inline Code/Engine/RendererCore/Debug/Implementation/DebugRenderer.cpp:391]: added an overall transform option

**Comment by jankrassnigg** [comment]
Related new documentation: http://ezengine.net/debugging/debug-rendering.html

File: Code/Engine/RendererCore/Debug/Implementation/DebugRenderer.cpp (hunk 1 above)
Comment (C-Core) [reply Code/Engine/RendererCore/Debug/Implementation/DebugRenderer.cpp:106]: Indeed, good catch!

File: Code/Engine/RendererCore/Debug/Implementation/DebugRenderer.cpp (hunk 2 above)
Comment (C-Core) [reply Code/Engine/RendererCore/Debug/Implementation/DebugRenderer.cpp:391]: makes sense

**Review by C-Core — APPROVED** [review]

**Pull request merged**